File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-0.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        // Determine the length of the new array
        int mergedLength = nums1.length + nums2.length;
        int[] mergedArray = new int[mergedLength];
        
        // Copy elements from nums1 to mergedArray
        System.arraycopy(nums1, 0, mergedArray, 0, nums1.length);
        
        // Copy elements from nums2 to mergedArray, starting after nums1 elements
        System.arraycopy(nums2, 0, mergedArray, nums1.length, nums2.length);

        // Sort the combined array O(k log k) where k = n+m
        Arrays.sort(mergedArray);

        if((mergedLength % 2) == 1) {
            return (double) mergedArray[mergedLength/2];
        }
        return ((double) mergedArray[mergedLength/2] + mergedArray[mergedLength/2 - 1])/2;


    }
}

File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        sort(s1.begin(), s1.end());
        int n1 = s1.size();
        int n2 = s2.size();
        for(int i=0; (i+n1)<n2; i++) {
            string sub = s2.substr(i, n1);
            sort(sub.begin(), sub.end());
            if (s1==sub)
                return true;
        }
        return false;
    }
};
